eurocodedesign/materials/structuralsteel.py

```diff
@@ -28,48 +28,69 @@
     shear_modulus = steel_material.G
     thermal_coefficient = steel_material.alpha
 """
-
+from abc import ABC, abstractmethod
 from dataclasses import dataclass, field
+from typing import Type
 
 
 @dataclass(frozen=True)
-class BasicStructuralSteel:
+class BasicStructuralSteel(ABC):
     _thickness_le_40mm: bool = True
     _elastic_modulus: float = field(default=210_000, kw_only=True)
     _shear_modulus: float = field(default=81_000, kw_only=True)
     _thermal_coefficient: float = field(default=1.2e-7, kw_only=True)  # 1/K
     poissons_ratio: float = field(default=0.3, kw_only=True)
 
     @property
-    def f_yk(self):
+    @abstractmethod
```

Comment:
You added the abstract methods so that when someone defines a new steel material they have to specifiy `_fy_thin` and so on right?

Reply:
Yes